api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java

```diff
@@ -131,4 +132,9 @@
      * @return the proxy config
      * */
     ProxyConfig getConfiguration();
+
+    /**
+     * Reloads proxy configuration and fires {@link ProxyReloadEvent}
+     */
+    void reloadConfiguration();
```

Comment:
Should this throw checked exception? Because then there's a way to force checking if reload failed (and if yes, then catching exception and figuring out what happened).

Reply:
I think it's better to not have this in the API.